tools/clang/lib/Analysis/UninitializedValues.cpp

```diff
@@ -35,8 +35,10 @@ using namespace clang;
 
 static bool isTrackedVar(const VarDecl *vd, const DeclContext *dc) {
   // HLSL Change Begin - Treat `out` parameters as uninitialized values.
-  if (vd->hasAttr<HLSLOutAttr>() && !vd->hasAttr<HLSLInAttr>())
-    return true;
+  if (vd->hasAttr<HLSLOutAttr>() && !vd->hasAttr<HLSLInAttr>()) {
+    QualType ty = vd->getType().getNonReferenceType();
+    return ty->isScalarType() || ty->isVectorType() || ty->isRecordType();
```

Comment:
If only thing we want to skip is array, is it possible to change to return !ty->isArrayType()?

Reply:
If you look a couple lines down, the analysis was only built for these types, so we shouldn't try to use it for more. In our case vectors and matrices are all record types anyways because of how we wrap them in class decls.